app/viewmodels/helpers/TimelineHelper.scala

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright 2023 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package viewmodels.helpers
+
+import models.response.emcsTfe.getMovementHistoryEvents.{GetMovementHistoryEventsResponse, MovementHistoryEvent}
+import play.api.i18n.Messages
+import utils.{DateUtils, Logging}
+import viewmodels.TimelineEvent
+
+import java.time.format.DateTimeFormatter
+import java.time.{LocalDate, LocalDateTime}
+import javax.inject.Inject
+import scala.util.Try
+
+class TimelineHelper @Inject()() extends Logging with DateUtils {
+
+  def timeline(historyEventsResponse: GetMovementHistoryEventsResponse)(implicit messages: Messages): Seq[TimelineEvent] =
+    historyEventsResponse.movementHistory.map { event =>
+      TimelineEvent(
+        eventType = event.eventType,
+        title = messages(getEventTitleKey(event)),
+        date = parseDateTime(event.eventDate),
+        url = getHistoryEventUrl(event)
+      )
+    }.sortBy(_.date)
```

Comment:
Should this also sort by `event.sequence` as a secondary sort?

Just thinking about the situation where multiple things happen on the same day. The date would be the same, so then would the next sort key make sense to be the `sequenceNumber` `asc`?

Or does this always have a timestamp entry on the date too? In which case, `dateTime` might be a better name for it I guess.

Reply:
It will always have a date/time so the sorting in order would be catered for, will change to `dateTime` to better reflect this.

Reply:
https://github.com/hmrc/emcs-tfe-frontend/pull/50/commits/eb11804ae81dec36fc5c8dec4aa1f618a3c4e288